src/attribute-editor/row.tsx

```diff
@@ -34,7 +34,10 @@ function render<T>(
   itemIndex: number,
   slot: AttributeEditorProps.FieldRenderable<T> | React.ReactNode | undefined
 ) {
-  return typeof slot === 'function' ? slot(item, itemIndex) : slot;
+  if (typeof slot === 'function') {
+    return (slot as AttributeEditorProps.FieldRenderable<T>)(item, itemIndex);
```

Comment:
From the PR description

> This PR removes unsafe usages of the any type

But I do not see any removed any's in this diff...

Reply:
If we asert the type of the return value, we have to keep that type assertion in sync with the definition of the return type of `FieldRenderable`, which sounds more risky to me than keeping only `FieldRenderable` in sync only across line 35 and line 38

Reply:
This is unsafe either way. We are not fixing any errors, just making the linter happy (which is another big discussion if these rules actually useful). 

I prefer the shorter option because it is more readable, cannot see any runtime hazards, really.

---

When looking into this PR I also learned where this strange `any` is coming from. `ReactNode` types are incorrect in our version. Related PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210, "Remove {} from ReactFragment" is the change we need

Reply:
Alternatively, we can write a separate typeguard function

```js
function isSlotFunction(slot: unknown) slot is AttributeEditorProps.FieldRenderable<T> {
   return typeof slot == 'function'
}

...

if (isSlotFunction(slot)) {
  return slot(item, itemIndex);
}
return slot;
```

This is also safe and readable to my standards